@xen-orchestra/proxy/src/app/mixins/api.js

```diff
@@ -53,7 +53,8 @@ export default class Api {
       try {
         body = parse(body)
       } catch (error) {
-        ctx.body = format.error(null, error)
+        warn('error on parsing request body', { error })
+        ctx.body = format.error(null, new JsonRpcError(error.message, error.code))
```

Comment:
Why is it necessary?

Reply:
According to the [`json-rpc-protocol`](https://github.com/JsCommunity/json-rpc-protocol#response) API,  the formatted errors should implement `toJsonRpcError`.

Reply:
Errors thrown by `json-rpc-protocol/parse` already do :slightly_smiling_face: